docs/design/2019-11-14-tidb-builtin-diagnostics-zh_CN.md

```diff
@@ -0,0 +1,680 @@
+
+火焰图快速定位问题的核心点是：
+
+- 提供全局视野
+- 展示全部调用路径
+- 层次化展示
+
+本提案提出的解决方案聚焦在解决核心问题的点上，而未拘泥于是图形展示形式。最终的方案为：对采样数据进行聚合，并将所有的调用路径使用树形结构逐行进行展示。
+
+解决方案是通过以下方式契合三个核心点：
+
+- 提供全局视野：对每一个聚合结果使用单独的一列展示在全局的使用比例，可以方便过滤排序
+- 展示全部调用路径：将所有的调用路径都作为查询结果，并使用单独的列对各个调用路径的子树进行编号，可以方便的通过过滤只查看某一个子树
+- 层次化展示：使用树形结构展示堆栈，使用单独的列记录栈的深度，可以方便的对不同栈的深度进行过滤
+
+本提案需要实现以下性能剖析表：
+
+
+| 表名 | 作用 |
+|------|-----|
+| tidb_profile_cpu | TiDB CPU 火焰图 |
+| tikv_profile_cpu | TiKV CPU 火焰图 |
+| tidb_profile_block | TiDB 阻塞情况火焰图 |
+| tidb_profile_memory | TiDB 内存对象火焰图 |
+| tidb_profile_allocs | 内存分配火焰图 |
+| tidb_profile_mutex | 锁的争用情况火焰图 |
+| tidb_profile_goroutines | 系统中已有的 goroutines，排查 goroutine 泄漏、阻塞 |
+
+#### 内存表全局化
+
+目前 `slow_query`/`events_statements_summary_by_digest`/`processlist` 只包含单节点数据，本提案通过添加以下三张集群级别系统表使任何一个 TiDB 实例可以查看整个集群的信息：
+
+| 表名 | 作用 |
+|------|-----|
+| tidb_cluster_slow_query | 所有 TiDB 节点的 slow_query 表数据 |
+| tidb_cluster_statements_summary | 所有 TiDB 节点的 statements summary 表数据 |
+| tidb_cluster_processlist | 所有 TiDB 节点的 processlist 表数据 |
+
+#### 所有节点的配置信息
+
+对于一个大集群，通过 HTTP API 去每一个节点获取配置的方式较为繁琐和低效，本提案提供全集群配置信息系统表，简化整个集群配置信息的获取、过滤、聚合。
+
+如下示例是实现本提案后的预期结果：
+
+```
+mysql> use information_schema;
+Database changed
+
+mysql> select * from tidb_cluster_config where `key` like 'log%';
++------+------+--------+-----------------+-----------------------------+---------------+
+| ID   | TYPE | NAME   | ADDRESS         | KEY                         | VALUE         |
++------+------+--------+-----------------+-----------------------------+---------------+
+|   21 | pd   | pd-0   | 127.0.0.1:2379  | log-file                    |               |
+|   22 | pd   | pd-0   | 127.0.0.1:2379  | log-level                   |               |
+|   23 | pd   | pd-0   | 127.0.0.1:2379  | log.development             | false         |
+|   24 | pd   | pd-0   | 127.0.0.1:2379  | log.disable-caller          | false         |
+|   25 | pd   | pd-0   | 127.0.0.1:2379  | log.disable-error-verbose   | true          |
+|   26 | pd   | pd-0   | 127.0.0.1:2379  | log.disable-stacktrace      | false         |
+|   27 | pd   | pd-0   | 127.0.0.1:2379  | log.disable-timestamp       | false         |
+|   28 | pd   | pd-0   | 127.0.0.1:2379  | log.file.filename           |               |
+|   29 | pd   | pd-0   | 127.0.0.1:2379  | log.file.log-rotate         | true          |
+|   30 | pd   | pd-0   | 127.0.0.1:2379  | log.file.max-backups        | 0             |
+|   31 | pd   | pd-0   | 127.0.0.1:2379  | log.file.max-days           | 0             |
+|   32 | pd   | pd-0   | 127.0.0.1:2379  | log.file.max-size           | 0             |
+|   33 | pd   | pd-0   | 127.0.0.1:2379  | log.format                  | text          |
+|   34 | pd   | pd-0   | 127.0.0.1:2379  | log.level                   |               |
+|   35 | pd   | pd-0   | 127.0.0.1:2379  | log.sampling                | <nil>         |
+|  114 | tidb | tidb-0 | 127.0.0.1:4000  | log.disable-error-stack     | <nil>         |
```

Comment:
tidb-0? can we name a TiDB instance?

Reply:
Actually, we cannot, it's a temporary name generated by `TIDB_CLUSTER_INFO`. Do we need to remote it?